destination/writer/batch.go

```diff
@@ -0,0 +1,188 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package writer
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/conduitio-labs/conduit-connector-snowflake/destination/schema/snowflake"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+type BatchType int
+
+const (
+	InsertBatch BatchType = iota + 1
+	UpdateBatch
+)
+
+type Batch struct {
```

Comment:
This struct was introduced to collect everything needed for a batch of records (i.e. a single CSV file). It can be created either using `NewInsertBatch` or `NewUpdateBatch` and depending on the function used it will produce a different merge query and a different filename.